app/controllers/User.scala

```diff
@@ -188,7 +191,9 @@ final class User(
                   "perfs"      -> lila.user.JsonView.perfsJson(user.perfs, user.perfs.best8Perfs)
                 )
           )
-      else Ok(views.user.bits.miniClosed(user.user))
+      else
+        val block = Await.result(relation.map(_.contains(Relation.Block)), Duration.Inf)
```

Comment:
we don't block threads. We compose asynchronous futures using `map`, `flatMap`, `flatMapN` etc